src/controllers/userController.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response } from 'express';
+import { User } from '../models/user';
+import { UserRepository } from '../repository/userRepository';
+import { BaseController } from './baseController';
+import { Validations } from '../validations';
+
+const checkRequest = new Validations;
+
+export class UserController extends BaseController {
+
+    userRepository: UserRepository;
+
+    constructor(userRepository: UserRepository) {
+        super();
+        this.userRepository = userRepository;
+    }
+    
+    signUp(req: Request, res: Response) {
+        const { email, password } = req.body;
+
+        checkRequest.validateEmailPassword(email, password)
+        .then((errors) => {
+            if (errors.length === 0) {
+                return this.userRepository.findUser(email)
+                .then((user) => {
+                    if (user === undefined) {
```

Comment:
you don't need to compare with `undefined`, just if (variable) works in this case.